File: src/main/java/com/group/libraryapp/domain/user/loanhistory/UserLoanHistoryRepository.java
package com.group.libraryapp.domain.user.loanhistory;

import com.group.libraryapp.domain.user.loan_history.UserLoanHistory;
import org.springframework.data.jpa.repository.JpaRepository;

public interface UserLoanHistoryRepository extends JpaRepository<UserLoanHistory, Long> {

  UserLoanHistory findByBookNameAndIsReturn(String bookName, boolean isReturn);

}

File: src/main/kotlin/com/group/libraryapp/domain/user/loan_history/UserLoanHistoryRepository.kt
package com.group.libraryapp.domain.user.loan_history

import org.springframework.data.jpa.repository.JpaRepository

interface UserLoanHistoryRepository: JpaRepository<UserLoanHistory,Long> {

    fun findByBookNameAndStatus(bookName: String, Status: UserLoanStatus): UserLoanHistory?

    fun countByStatus(status: UserLoanStatus): Long
}